cds/webapi/C#/UploadDownloadFilesImagesUsingWebAPI/UploadDownloadFilesImagesUsingWebAPI/WebApiBlobUploadDownload.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Microsoft.Xrm.Tooling.Connector;
+using PowerApps.Samples.UploadDownloadFiles;
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+
+namespace PowerApps.Samples
+{
+    public class WebApiBlobUploadDownload
+    {
+        CrmServiceClient _service = null;
+      
+        public WebApiBlobUploadDownload(CrmServiceClient service)
+        {
+            _service = service;
+        }
+
+        /// <summary>
+        /// Uploads a file or image using Web API
+        /// </summary>
+        /// <param name="entityPluralName">entity name, such as accounts</param>
+        /// <param name="accountId">an account record guid id</param>
+        /// <param name="logicalName">a customized column logical name</param>
+        /// <param name="blobName">a file or image to be uploaded</param>
+        public void BlobUploadUsingAPI(string entityPluralName, Guid accountId, string logicalName, string blobName)
+        {
+            try
+            {
+                if (!_service.IsReady)
+                {
+                    Console.WriteLine($"Crm Service is not ready to use");
+                    return;
+                }
+
+                var accessToken = _service.CurrentAccessToken;
+                var blobUrl = $"{Constants.webApiUrl}/{entityPluralName}({accountId})/{logicalName}";
+                var blobFullPath = Path.GetFullPath(blobName).Replace("\\bin\\Debug", "");
+                var uploadBlobName = Path.GetFileName(blobFullPath);
+                var fileStream = File.OpenRead(blobFullPath);
+
+                var request = new HttpRequestMessage()
+                {
+                    Method = new HttpMethod("Patch"),
+                    RequestUri = new Uri(blobUrl)
+                };
+                request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
+                request.Content = new StreamContent(fileStream);
+                request.Content.Headers.Add("Content-Type", "application/octet-stream");
+                request.Content.Headers.Add("x-ms-file-name", uploadBlobName);
+
+                var httpClient = new HttpClient();
```

Comment:
It is strange to me that this sample depends so much on CrmServiceClient, but uses it's own HttpClient.
CrmServiceClient exposes [CrmServiceClient.ExecuteCrmWebRequest Method](https://docs.microsoft.com/en-us/dotnet/api/microsoft.xrm.tooling.connector.crmserviceclient.executecrmwebrequest?view=dataverse-sdk-latest) that accepts method, headers and body and returns an HttpResponseMessage.
If we are going to depend on CrmServiceClient, why not go 'all-in'?

I think that this sample should use CDSWebAPIService to be consistent with all our other Web API samples.